apollo-router/src/query_planner/fetch.rs

```diff
@@ -243,6 +255,189 @@ impl Display for QueryHash {
 pub(crate) struct Variables {
     pub(crate) variables: Object,
     pub(crate) inverted_paths: Vec<Vec<Path>>,
+    pub(crate) contextual_args: Option<(HashSet<String>, usize)>,
+}
+
+#[derive(Debug)]
+enum ContextBatchingError {
+    InvalidDocument(WithErrors<Document>),
```

Comment:
If the changes @lrlna suggests about not re-parsing the operation are adopted, then this enum variant isn't required. With only one variant, it may even make sense to convert this to a struct.